pageserver/src/tenant.rs

```diff
@@ -1405,17 +1414,26 @@ impl Tenant {
             offloaded_timelines_list.push((timeline_id, Arc::new(offloaded_timeline)));
             offloaded_timeline_ids.insert(timeline_id);
         }
+        // Complete deletions for offloaded timeline id's.
+        offloaded_timelines_list
+            .retain(|(offloaded_id, _offloaded)| {
+                // In the end, existence of a timeline is finally determined by the existence of an index-part.json in remote storage.
+                // If there is a dangling reference in another location, they need to be cleaned up.
+                let delete = !preload.offloaded_with_prefix.contains(offloaded_id);
+                if delete {
+                    tracing::info!("Removing offloaded timeline {offloaded_id} from manifest as no remote prefix was found");
+                }
```

Comment:
I think the log line is enough.

```suggestion
                let delete = !preload.offloaded_with_prefix.contains(offloaded_id);
                if delete {
                    tracing::info!("Removing offloaded timeline {offloaded_id} from manifest as no remote prefix was found");
                }
```

Reply:
I think spelling this out is useful because you might know the content, but not someone else who reads it, and it's non-obvious.

Reply:
Your call